src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs

```diff
@@ -1312,5 +1312,166 @@ public void Match_Wordchar()
             }
             Assert.Empty(ambiguous);
         }
+
+        [Theory]
+        [InlineData(RegexOptions.None)]
+        [InlineData(RegexOptions.Compiled)]
+        [InlineData(RegexHelpers.RegexOptionNonBacktracking)]
+        public void StressTestDeepNestingOfConcat(RegexOptions options)
+        {
+            if (PlatformDetection.IsNetFramework && options == RegexHelpers.RegexOptionNonBacktracking)
+            {
+                return;
+            }
```

Comment:
Could we move the RegexOptionsExtended and RegexOptions_TestData into a shared test class (I believe we already have one where we put helpers)?  Then this can just use that same MemberData, and we can get rid of some duplication elsewhere as well.

Reply:
It turns out that `MemberData` can only reference a method in the same class where the test itself is defined, so it seems moving `RegexOptions_TestData` to `RegexHelpers` class is not allowed. I'll move `RegexOptionsExtended`.

Reply:
> It turns out that MemberData can only reference a method in the same class where the test itself is defined

You can set a MemberType property on the attribute, e.g.
https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Linq.Parallel/tests/QueryOperators/OfTypeTests.cs#L33

Reply:
Good to know. Thanks! I'll use that attribute.